src/main/java/me/tomsdevsn/hetznercloud/UrlBuilder.java

```diff
@@ -0,0 +1,47 @@
+package me.tomsdevsn.hetznercloud;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class UrlBuilder {
+
+    private String url;
+
+    private Map<String, String> queryParams = new HashMap<>();
+
+    private UrlBuilder(String url) {
+        this.url = url;
+    }
+
+    public static UrlBuilder from(String url) {
+
+        if (url == null || url.isBlank()) {
+            throw new RuntimeException("url must not be empty or blank");
+        }
+
+        return new UrlBuilder(url);
+    }
+
+    public UrlBuilder queryParamIfPresent(String name, Optional<Integer> value) {
+        value.ifPresent((it) -> queryParams.put(name, it.toString()));
+        return this;
```

Comment:
If I'm not wrong you don't need to do `return this` and return the `value.ifPresent` directly.

Reply:
return this was intentional to allow for fluent builder like construction of urls